config/samples/resources/alloydbinstance/primary-instance/alloydb_v1beta1_alloydbinstance.yaml

```diff
@@ -23,5 +23,11 @@ spec:
     name: alloydbinstance-dep-primary
   databaseFlags:
     enable_google_adaptive_autovacuum: "off"
+    password.enforce_complexity: "on"
```

Comment:
Thanks for updating this field example.
I think this field should be changed to `passwordPolicyFlags`, with struct value `enforce_complexity`, `expiration_in_days `, etc 

```
passwordPolicyFlags:
    enforce_complexity: "on"
    expiration_in_days: "90"
```

Also not a blocker.

Reply:
`databaseFlags` is a freeform map field so I think the value will be whatever the API takes. And looks like `password.enforce_complexity` is exactly the value the API asks the user to pass in: https://cloud.google.com/alloydb/docs/manage-password-policy#enforce-password-complexity

I personally also don't completely agree with the structure/format, but I guess this is an API decision that's already GAed, and any improvement of it is out of the scope for this PR.

Reply:
Yes, like Joyce said we have to use `password.enforce_complexity` as the flag name - I think it'd be a bit difficult for us to change the AlloyDB API format with the database flags at this point unfortunately

Reply:
Okay, as a FYI, this could block users to use some popular k8s tools and block configconnector to correctly populate the field value in certain cases. See
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#selecting-fields

Reply:
Hi @yuwenma, do you mean that the field name `password.enforce_complexity` may be invalid for some K8s tools so that users cannot select this field in certain use cases?

Reply:
> Okay, as a FYI, this could block users to use some popular k8s tools and block configconnector to correctly populate the field value in certain cases. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#selecting-fields

Thanks for letting me know! This is surprising, we have other flags that follow this same format of having a period in the flag name e.g. [`google_columnar_engine.auto_columnarization_schedule`](https://cloud.google.com/alloydb/docs/reference/columnar-engine-flags#auto-columnarization-schedule) and [`google_db_advisor.max_statement_length`](https://cloud.google.com/alloydb/docs/reference/index-advisor-flags#max_statement_length)

CloudSQL which also has KCC support I think does a similar pattern for their db flags: https://cloud.google.com/sql/docs/postgres/flags#list-flags-postgres. I'd expect this to be a problem for them too

Reply:
Yes, I think so. https://kubernetes.io/docs/reference/kubectl/jsonpath (if users use json path)